packages/@aws-cdk/aws-ec2/lib/launch-template.ts

```diff
@@ -0,0 +1,707 @@
+    const haveName = Boolean(attrs.launchTemplateName);
+    if (haveId == haveName) {
+      throw new Error('LaunchTemplate.fromLaunchTemplateAttributes() requires exactly one of launchTemplateId or launchTemplateName be provided.');
+    }
+
+    class Import extends Resource implements ILaunchTemplate {
+      public readonly versionNumber = attrs.versionNumber ?? LaunchTemplateSpecialVersions.DEFAULT_VERSION;
+      public readonly launchTemplateId? = attrs.launchTemplateId;
+      public readonly launchTemplateName? = attrs.launchTemplateName;
+    }
+    return new Import(scope, id);
+  }
+
+  // ============================================
+  //   Members for ILaunchTemplate interface
+
+  public readonly versionNumber: string;
+  public readonly launchTemplateId?: string;
+  public readonly launchTemplateName?: string;
+
+  // =============================================
+  //   Data members
+
+  /**
+   * The default version for the launch template.
+   *
+   * @attribute
+   */
+  public readonly defaultVersionNumber: string;
+
+  /**
+   * The latest version of the launch template.
+   *
+   * @attribute
+   */
+  public readonly latestVersionNumber: string;
+
+  /**
+   * TagManager for tagging support.
+   */
+  protected readonly tags: TagManager;
+
+  // =============================================
+  //   Private/protected data members
+
+  /**
+   * The type of OS the instance is running.
+   * @internal
+   */
+  protected readonly _osType?: OperatingSystemType;
+
+  /**
+   * IAM Role assumed by instances that are launched from this template.
+   * @internal
+   */
+  protected readonly _role?: iam.IRole;
+
+  /**
+   * Principal to grant permissions to.
+   * @internal
+   */
+  protected readonly _grantPrincipal?: iam.IPrincipal;
+
+  /**
+   * Allows specifying security group connections for the instance.
+   * @internal
+   */
+  protected readonly _connections?: Connections;
+
+  /**
+   * UserData executed by instances that are launched from this template.
+   * @internal
+   */
+  protected readonly _userData?: UserData;
+
+  // =============================================
+
+  constructor(scope: Construct, id: string, props: LaunchTemplateProps = {}) {
+    super(scope, id);
+
+    // Basic validation of the provided spot block duration
+    const spotDuration = props?.spotOptions?.blockDuration?.toHours({ integral: true });
+    if (spotDuration !== undefined && (spotDuration < 1 || spotDuration > 6)) {
+      // See: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-requests.html#fixed-duration-spot-instances
+      Annotations.of(this).addError('Spot block duration must be exactly 1, 2, 3, 4, 5, or 6 hours.');
+    }
+
+    this._role = props.role;
+    this._grantPrincipal = this._role;
+    const iamProfile: iam.CfnInstanceProfile | undefined = this._role ? new iam.CfnInstanceProfile(this, 'Profile', {
+      roles: [this._role!.roleName],
+    }) : undefined;
+    const iamProfileToken = Lazy.string({
```

Comment:
This feels like it doesn't need to be  lazy?

```ts
const iamProfileToken = iamProfile?.getAtt('Arn').toString();
```
?

Or if you MUST have that the type will be `string` instead of `string | undefined` (although why would you):

```ts
const iamProfileToken = iamProfile ? iamProfile.getAtt('Arn').toString() : Token.asString(undefined);
```

Reply:
Yeah, I got a little carried away with the `Lazy` uses :-)